eng/scripts/Get-ApiViewCommandLine.ps1

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+# Usage: Get-ApiViewCommandLine.ps1 .\azure-sdk-for-cpp azure-identity
+# Or: ..\Get-ApiViewCommandLine.ps1 . azure-core
+# Or: Get-ApiViewCommandLine.ps1 ..\.. azure-security-attestation
+# Or: c:\src\azure-sdk-for-cpp\eng\scripts\Get-ApiViewCommandLine.ps1 c:\src\azure-sdk-for-cpp azure-identity
+
+param([String]$RepoPath, [String]$LibName)
+
+[String]$SdkRoot = Resolve-Path ($RepoPath + "\sdk")
+
+[String[]]$AllIncDirs = Get-ChildItem -Directory -Filter "inc" -Recurse $SdkRoot | Select-Object -ExpandProperty FullName
+
+[String[]]$AllIncDirsWithoutInc = $AllIncDirs | Select-Object @{ Label="Substr"; Expression = { $_.Substring(0, $_.Length - "inc".Length) } } | Select-Object -ExpandProperty Substr
+
+[String[]]$AllLibIncDirs = @()
+for($i = 0; $i -lt $AllIncDirsWithoutInc.Length; $i++) {
+    $isLibDir = $true
+    $libDir = $AllIncDirsWithoutInc[$i]
+    for($j = 0; $j -lt $AllIncDirsWithoutInc.Length; $j++) {
+        if ($i -eq $j) {
+            continue
+        }
+
+        $StartsWith = $AllIncDirsWithoutInc[$j] + "*"
+        if ($libDir -Like $StartsWith) {
+            $isLibDir = $false
+            break
+        }
+    }
+
+    if ($isLibDir){
+        $AllLibIncDirs += $libDir + "inc"
+    }
+}
+
+[String]$LibIncDir = $AllLibIncDirs | Where-Object {$_ -Match ("\\" + $LibName + "\\inc") } | Select-Object -First 1
+
+[String[]]$LibHeaders = Get-ChildItem -File -Recurse $LibIncDir | Select-Object -ExpandProperty FullName
+
+$CmdLine = "clang++"
+foreach ($header in $LibHeaders) {
+    $CmdLine += " " + $header
+}
+
+$CmdLine += " -Xclang -ast-dump"
+
+foreach ($incDir in $AllLibIncDirs) {
+    $CmdLine += " -I " + $incDir
+}
+
+$CmdLine
```

Comment:
Do we want this to just return a command line? Why not have it execute the command line?

Reply:
This is more of a building block, of somewhat temporary shape, for us to sync with Praveen, and integrate ApiView with our repo one way or another. I do not know the exact output type that is expected, but it can be, and most probably will be changed the way you suggest. Maybe even to an extent that it would execute command line, write AST tree to a file, zip it, and rename the archive, i.e. will do all the ApiView's instructions for C++. The rest should be relatively easy, it is the right set of files and directories that is C++ specific and for which Praveen needed someone from the C++ team to figure out.
Currently, it prints the result, which is the easiest way to find out the ouput and, make sure it works, and debug if necessary.